src/test/scala/io/iohk/ethereum/network/NodeParserSpec.scala

```diff
@@ -42,4 +40,33 @@ class NodeParserSpec extends FlatSpec with Matchers with PropertyChecks {
     }
   }
 
+  it should "correctly parse IPv6 nodes" in {
+    val testVectors = Table[String, Option[String]](
+      ("nodes", "expectedOutput"),
+
+      ("enode://c94b6f71c2f3d84ed5587ff936172138cfd4af4951e4ca784b9ea5330f76ed8d77d23a7178b18716947a17a8ef59f18519bc0064e7f3f12e0c1c5934cac147a0@[ce90:c2c:7000:0:10:0:0:0]:30303",
+        Some("enode://c94b6f71c2f3d84ed5587ff936172138cfd4af4951e4ca784b9ea5330f76ed8d77d23a7178b18716947a17a8ef59f18519bc0064e7f3f12e0c1c5934cac147a0@[ce90:c2c:7000:0:10:0:0:0]:30303")
+      ), //Has full IPv6 address
+      ("enode://c94b6f71c2f3d84ed5587ff936172138cfd4af4951e4ca784b9ea5330f76ed8d77d23a7178b18716947a17a8ef59f18519bc0064e7f3f12e0c1c5934cac147a0@[ce90:c2c:7000:0:10::]:30303",
+        Some("enode://c94b6f71c2f3d84ed5587ff936172138cfd4af4951e4ca784b9ea5330f76ed8d77d23a7178b18716947a17a8ef59f18519bc0064e7f3f12e0c1c5934cac147a0@[ce90:c2c:7000:0:10:0:0:0]:30303")
+      ), //Has partial IPv6 address
+      ("enode://c94b6f71c2f3d84ed5587ff936172138cfd4af4951e4ca784b9ea5330f76ed8d77d23a7178b18716947a17a8ef59f18519bc0064e7f3f12e0c1c5934cac147a0@ip6-localhost:30303",
+        Some("enode://c94b6f71c2f3d84ed5587ff936172138cfd4af4951e4ca784b9ea5330f76ed8d77d23a7178b18716947a17a8ef59f18519bc0064e7f3f12e0c1c5934cac147a0@ip6-localhost:30303")
```

Comment:
This test case is failing for me locally.
https://github.com/input-output-hk/mantis/pull/311/files#diff-f690d49aebd7d49500ebae9e4965e476L54 throw exception `ip6-localhost: nodename nor servname provided, or not known` and this 
`node.isRight shouldEqual maybeExpectedOutput.nonEmpty` assertion fails, as `node` is `Left(Set(ip6-localhost: nodename nor servname provided, or not known))`

Reply:
It seems to be something OS dependent (this test also passed locally on my computer), the alias `ip6-localhost` is not universal.
I just pushed a commit to fix this, nice catch :+1: